src/controller/SetUpCodePairer.cpp

```diff
@@ -304,9 +304,6 @@ CHIP_ERROR SetUpCodePairer::StartDiscoveryOverWiFiPAF()
 CHIP_ERROR SetUpCodePairer::StopDiscoveryOverWiFiPAF()
 {
     mWaitingForDiscovery[kWiFiPAFTransport] = false;
-#if CHIP_DEVICE_CONFIG_ENABLE_WIFIPAF
-    DeviceLayer::ConnectivityMgr().WiFiPAFCancelIncompleteSubscribe();
```

Comment:
So this seems like the heart of the change, right?

Trying to make sure I understand this: what prevents use-after-free here?  We passed a pointer to `this` to `WiFiPAFSubscribe`.  If we are here, `this` might be about to become deallocated.  What prevents PAF from still using the pointer after this?